chainer/functions/pooling/max_pooling_2d.py

```diff
@@ -158,7 +158,7 @@ def forward_cpu(self, gy):
             (n * c * out_h * out_w * kh * kw), dtype=self._in_dtype)
 
         indexes = self.indexes.flatten()
-        indexes += numpy.arange(0, indexes.size * kh * kw, kh * kw)
+        indexes = indexes + numpy.arange(0, indexes.size * kh * kw, kh * kw)
```

Comment:
`indexes = self.indexes.ravel() + numpy.arange(0, indexes.size * kh * kw, kh * kw)` looks better because we no longer worry about overwriting `self.indexes`.

Reply:
yes.